engine/src/main/java/org/terasology/engine/core/module/ModuleManager.java

```diff
@@ -128,40 +130,61 @@ private void loadModulesFromApplicationPath(PathManager pathManager) {
         scanner.scan(registry, paths);
     }
 
-    private Module loadEngineModule(List<Class<?>> classesOnClasspathsToAddToEngine) {
+    /**
+     * Load and configure the engine module.
+     * <p>
+     * The engine module is the parts of the engine which are available to be called directly
+     * from other modules. Unlike other modules, engine classes are on the classpath and not
+     * restricted by the ModuleClassLoader.
+     * <p>
+     * This function is static so it can be tested without needing a ModuleManager instance.
+     *
+     * @param moduleFactory used to create the module
+     * @param classesOnClasspathsToAddToEngine added to the module's reflections manifest
+     */
+    static Module loadAndConfigureEngineModule(ModuleFactory moduleFactory, List<Class<?>> classesOnClasspathsToAddToEngine) {
+        // Start by creating a gestalt Module for the Java package `org.terasology.engine`.
         Module packageModule = moduleFactory.createPackageModule("org.terasology.engine");
 
         // We need to add reflections from our subsystems and other classes.
-        ConfigurationBuilder config = new ConfigurationBuilder();
         Reflections packageReflections = packageModule.getModuleManifest();
-        Set<Scanner> scanners = packageReflections.getConfiguration().getScanners();
-        config.setScanners(scanners.toArray(new Scanner[0]));
+        ConfigurationBuilder config = reflectionsConfigurationFrom(packageReflections);
 
+        Collection<File> classPaths = new HashSet<>(packageModule.getClasspaths());
         for (Class<?> aClass : classesOnClasspathsToAddToEngine) {
-            config.addUrls(ClasspathHelper.forClass(aClass));
+            URL url = ClasspathHelper.forClass(aClass);
+            config.addUrls(url);  // include this in reflections scan
+            classPaths.add(urlToFile(url));  // also include in Module.moduleClasspaths
         }
 
-        // TODO: is this using reflections.cache?
-        Reflections reflectionsWithSubsystems = new Reflections(config);
-        packageReflections.merge(reflectionsWithSubsystems);
+        if (!config.getUrls().isEmpty()) {
+            Reflections reflectionsWithSubsystems = new Reflections(config);
+            packageReflections.merge(reflectionsWithSubsystems);
+        }
 
         // We need the class predicate to include classes in subsystems and whatnot. We can't change it in an
-        // existing module, so make a new one based on the one from the moduleFactory.
-        Module engine = new Module(
+        // existing module, so make a new one based on the one from the factory.
+        // TODO: expand the ModuleFactory interface to make this whole thing less awkward
+        return new Module(
                 packageModule.getMetadata(),
                 packageModule.getResources(),
-                Collections.emptyList(),
+                classPaths,
```

Comment:
This is potentially a functional change: engine's `Module.moduleClasspaths` is no longer an empty list.

Reply:
It doesn't seem to make a difference for the Malicious results. At least not in a local dev workspace running under gradle, I haven't ruled out the possibility that it's useful for something in other configurations.